src/tools/illink/test/Mono.Linker.Tests.Cases/LinkXml/TypeWithPreserveFieldsHasBackingFieldsOfPropertiesRemoved.cs

```diff
@@ -62,6 +62,7 @@ interface IFoo3<T, K, J>
 			int Bar3 { get; set; }
 		}
 
+		[Kept (By = Tool.NativeAot)]
```

Comment:
This test is not actually testing anything relevant, so I'm not worried about "regression" here.

* In ILLinker, looks like `preserve=fields` is going to skip over compiler generated backing fields. Not sure where that logic is but we clearly don't do that in native AOT and we do preserve them. Won't lose my sleep over that.
* The test infrastructure doesn't check field preservation on native AOT side since we don't really "trim" fields (we vacate space for them no matter what, there's leeway whether reflection metadata is generated).

So the only observable effect in this test is whether the type of the field survived and now it by design survives if the field was a target of reflection.